app/src/main/java/com/tari/android/wallet/util/SharedPrefsWrapper.kt

```diff
@@ -41,59 +41,88 @@ import android.content.SharedPreferences
  */
 class SharedPrefsWrapper(private val sharedPrefs: SharedPreferences) {
 
-    private val privateKeyHexStringKey = "tari_wallet_private_key_hex_string"
-    private val onboardingStartedKey = "tari_wallet_onboarding_started"
-    private val onboardingCompletedKey = "tari_wallet_onboarding_completed"
-    private val onboardingDisplayedAtHome = "tari_wallet_onboarding_displayed_at_home"
-
-    fun getPrivateKeyHexString(): String? {
-        return sharedPrefs.getString(privateKeyHexStringKey, null)
+    private object Key {
+        const val privateKeyHexStringKey = "tari_wallet_private_key_hex_string"
+        const val publicKeyHexStringKey = "tari_wallet_public_key_hex_string"
+        const val emojiIdKey = "tari_wallet_emoji_id_"
+        const val onboardingStartedKey = "tari_wallet_onboarding_started"
+        const val onboardingCompletedKey = "tari_wallet_onboarding_completed"
+        const val onboardingDisplayedAtHomeKey = "tari_wallet_onboarding_displayed_at_home"
     }
 
-    fun setPrivateKeyHexString(privateKeyHexString: String?) {
-        sharedPrefs.edit().apply {
-            putString(privateKeyHexStringKey, privateKeyHexString)
-            apply()
+    var privateKeyHexString: String?
```

Comment:
Question:

i) How secure is SharedPreferences?

Reply:
Never tried but it is said that it can be accessed by anyone who has root access, which makes sense. So I opened up #125 for it - encrypt the key before storing it in the shared preferences. @CjS77

Reply:
Ok, cool. We have exactly the same issue for the desktop wallet and base node, so it's a fast-follow feature.

Reply:
Latest commit stores the private key securely after enryption using [1.0.0 release](https://mvnrepository.com/artifact/de.adorsys.android/securestoragelibrary/1.0.0) of the library [Secure Device Storage - Android](https://github.com/adorsys/secure-storage-android).